services/controller/standard/events.go

```diff
@@ -332,3 +337,41 @@ func (s *Service) subscribeToBeaconCommittees(ctx context.Context,
 	s.subscriptionInfos[epoch] = subscriptionInfo
 	s.subscriptionInfosMutex.Unlock()
 }
+
+// VerifySyncCommitteeMessages handles the "head" events from the beacon node for sync committee validation.
+func (s *Service) VerifySyncCommitteeMessages(ctx context.Context, data *apiv1.HeadEvent) {
+	_, span := otel.Tracer("attestantio.vouch.services.controller.standard").Start(ctx, "VerifySyncCommitteeMessages")
+	defer span.End()
+
+	logger := log.With().Uint64("slot", uint64(data.Slot)).Logger()
+	logger.Trace().Msg("Received head event")
+
+	rootReported, found := s.syncCommitteeMessenger.GetBeaconBlockRootReported(data.Slot)
```

Comment:
I don't think that this is the correct logic.  In slot `x` the sync committee message is generated using the head of the chain, which is usually the root of block `x`.  So in slot `x+1` (i.e. here) we want to look for the data for the previous slot, and compare it to the head block's parent hash.